### ISSUE TITLE
Add lint for match `io::ErrorKind::Other`

### ISSUE THREAD

**Issue by Xuanwo** [rust-lang/rust-clippy#9004]
### What it does

`io::ErrorKind::Other` is not recommended to be used in the match:

> A custom error that does not fall under any other I/O error kind.
>
> This can be used to construct your own [Error](https://doc.rust-lang.org/stable/std/io/struct.Error.html)s that do not match any [ErrorKind](https://doc.rust-lang.org/stable/std/io/enum.ErrorKind.html).
>
> This [ErrorKind](https://doc.rust-lang.org/stable/std/io/enum.ErrorKind.html) is not used by the standard library.
>
> Errors from the standard library that do not fall under any of the I/O error kinds cannot be matched on, and will only match a wildcard (_) pattern. New [ErrorKind](https://doc.rust-lang.org/stable/std/io/enum.ErrorKind.html)s might be added in the future for some of those.

But `Other` is easy to be used wrongly.

It's better for us to warn of this usage.

As proposed in https://github.com/rust-lang/rust/issues/86442#issuecomment-1157442038

### Lint Name

match_io_error_other

### Category

correctness

### Advantage

- Avoid breaking after the new `ErrorKind` variant introduced

### Drawbacks

- None

### Example

```rust
let Err(e) = std::fs::rename("a", "b") {
    match e {
        std::io::ErrorKind::Other => {}
    }
}
```

Could be written as:

```rust
let Err(e) = std::fs::rename("a", "b") {
    match e {
        _ => {}
    }
}
```
